Added solution for Project Euler problem 72

fpringle opened the pull request:
Name: Counting fractions

Consider the fraction, n/d, where n and d are positive integers. If n<d and HCF(n,d)=1, it is called a reduced proper fraction.

If we list the set of reduced proper fractions for d ≤ 8 in ascending order of size, we get:

1/8, 1/7, 1/6, 1/5, 1/4, 2/7, 1/3, 3/8, 2/5, 3/7, 1/2, 4/7, 3/5, 5/8, 2/3, 5/7, 3/4, 4/5, 5/6, 6/7, 7/8

It can be seen that there are 21 elements in this set.

How many elements would be contained in the set of reduced proper fractions for d ≤ 1,000,000?

Reference: [https://projecteuler.net/problem=72](url)
Reference: #2695


* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

dhruvmanila commented on project_euler/problem_72/sol1.py:
```diff
@@ -0,0 +1,43 @@
+"""
+Consider the fraction, n/d, where n and d are positive integers. If n<d and HCF(n,d)=1,
```
```suggestion
"""
Project Euler Problem 72: https://projecteuler.net/problem=72

Consider the fraction, n/d, where n and d are positive integers. If n<d and HCF(n,d)=1,
```

dhruvmanila commented on project_euler/problem_72/sol1.py:
```diff
@@ -0,0 +1,43 @@
+
+
+if __name__ == "__main__":
+    print(solution())
```
```suggestion
    print(f"{solution() = }")
```

dhruvmanila commented on project_euler/problem_72/sol1.py:
```diff
@@ -0,0 +1,43 @@
+    primes = set(range(3, limit, 2))
+    primes.add(2)
+    for p in range(3, limit, 2):
+        if p not in primes:
+            continue
```
```suggestion
```
These two lines are not needed.

fpringle replied:
This is an optimisation - there's no point in removing the factors of p from primes if p isn't prime, as all its factors will already have been eliminated. 

Reference: [Sieve of Eratosthenes](https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes#Pseudocode)

    algorithm Sieve of Eratosthenes is
        let A be an array of Boolean values, indexed by integers 2 to n,
        initially all set to true.
        
        for i = 2, 3, 4, ..., not exceeding √n do
            if A[i] is true           << here
                for j = i2, i2+i, i2+2i, i2+3i, ..., not exceeding n do
                    A[j] := false
    
        return all i such that A[i] is true.

[Here's to a script I wrote that defines functions to generate primes with and without the optimisation, showing a speedup in that section of the code by about 50%.](https://pastebin.com/a1XQCjuk)

dhruvmanila replied:
Ah, I'm sorry I totally missed the part where you were updating the `set`.

dhruvmanila reviewed (commented):
### Announcement:

@fpringle

*This message is sent to all pull requests which are **OPEN, NOT MERGED, not containing 'wontfix' label and contains 'Project Euler' or 'Project' or 'Euler' (case doesn't matter) in their title**. If this message doesn't apply to your pull request, please ignore this.*

### Message:

This is to notify all the users submitting a pull request for the Project Euler solution that the [README.md](https://github.com/TheAlgorithms/Python/blob/master/project_euler/README.md) file has been updated with all the style requirements necessary for the directory and solution files. ***Please read through it and make all the necessary changes required to your submission.***

dhruvmanila commented:
Um, you just deleted the old problem and replaced it with your own. As of now, we are only accepting solutions to problems that do not exist in this repository and you need to stop making multiple pull request. Please only submit one PR at a time.

fpringle commented:
> Um, you just deleted the old problem and replaced it with your own. As of now, we are only accepting solutions to problems that do not exist in this repository and you need to stop making multiple pull request. Please only submit one PR at a time.

My solution was added 6 days ago, the solution that I just deleted was added 3 days ago but merged before mine.
[Commit history](https://github.com/TheAlgorithms/Python/tree/2d7e08ef8324867f47f381cdc800ee00f6d77be0/project_euler/problem_72)

dhruvmanila commented:
Then please change the file name for your solution to `sol2.py`

dhruvmanila commented:
You're deleting the first solution. Please revert it back.

fpringle commented:
Apologies for the hastle, I'm new to contributing!

dhruvmanila commented:
You follow the guidelines perfectly well and so it is easier to review :) 
Keep up the good work. From next time, make sure to open ***only one PR*** at a time.

fpringle commented:
> From next time, make sure to open _**only one PR**_ at a time.

Sure thing - I can see now why multiple pull requests can be annoying :rofl:
